pkg/issuer/acme/http/service_test.go

```diff
@@ -162,6 +162,238 @@ func TestEnsureService(t *testing.T) {
 				}
 			},
 		},
+		"http-01 ingress challenge without a service type should default to NodePort": {
+			Challenge: &cmacme.Challenge{
+				Spec: cmacme.ChallengeSpec{
+					DNSName: "test.com",
+					Solver: cmacme.ACMEChallengeSolver{
+						HTTP01: &cmacme.ACMEChallengeSolverHTTP01{
+							Ingress: &cmacme.ACMEChallengeSolverHTTP01Ingress{},
+						},
+					},
+				},
+			},
+			PreFn: func(t *testing.T, s *solverFixture) {
+				expectedService, err := buildService(s.Challenge)
+				if err != nil {
+					t.Errorf("expectedService returned an error whilst building test fixture: %v", err)
+				}
+				// create a reactor that fails the test if a service is created
+				s.Builder.FakeKubeClient().PrependReactor("create", "services", func(action coretesting.Action) (handled bool, ret runtime.Object, err error) {
+					service := action.(coretesting.CreateAction).GetObject().(*v1.Service)
+					// clear service name as we don't know it yet in the expectedService
+					service.Name = ""
+					if !reflect.DeepEqual(service, expectedService) {
+						t.Errorf("Expected %v to equal %v", service, expectedService)
+					}
+					return false, ret, nil
+				})
+
+				s.Builder.Sync()
+			},
+			CheckFn: func(t *testing.T, s *solverFixture, args ...interface{}) {
+				resp := args[0].(*v1.Service)
+				err := args[1]
+				if resp == nil && err == nil {
+					t.Errorf("unexpected service = nil")
+					t.Fail()
+					return
+				}
+				services, err := s.Solver.serviceLister.List(labels.NewSelector())
+				if err != nil {
+					t.Errorf("unexpected error listing services: %v", err)
+					t.Fail()
+					return
+				}
+				if len(services) != 1 {
+					t.Errorf("unexpected %d services in lister: %+v", len(services), services)
+					t.Fail()
+					return
+				}
+				if !reflect.DeepEqual(services[0], resp) {
+					t.Errorf("Expected %v to equal %v", services[0], resp)
+				}
+				if services[0].Spec.Type != v1.ServiceTypeNodePort {
+					t.Errorf("Blank service type should default to NodePort, but was \"%s\"", services[0].Spec.Type)
```

Comment:
question / nitpick(non-blocking): would `%q` be better here? or is there a reason for `\"%s\"`?

same goes elsewhere in this file

```suggestion
					t.Errorf("Blank service type should default to NodePort, but was %q", services[0].Spec.Type)
```